src/v2/controllers/FormController.js

```diff
@@ -151,41 +150,39 @@ export default Controller.extend({
   },
 
   handleSaveForm(model) {
+    const currentViewState = this.options.appState.getCurrentViewState();
     const formName = model.get('formName');
 
-    const idx = this.options.appState.get('idx');
-    if (!idx['neededToProceed'].find(item => item.name === formName)) {
-      this.options.settings.callGlobalError(`Cannot find http action for "${formName}".`);
-      this.showFormErrors(this.formView.model, 'Cannot find action to proceed.');
-      return;
-    }
-
+    // Toggle Form saving status (e.g. disabling save button, etc)
     this.toggleFormButtonState(true);
     model.trigger('request');
 
+    // May need to redirect instead of AJAX request.
     // NOTE: okta-idx-js does not know whether a remediation
     // is required to redirect or perform a GET request.
     // As a result, we handle this accordingly by checking the formName
-    const fullPageRedirectForms = [ FORMS.REDIRECT_IDP ];
-    if (fullPageRedirectForms.includes(formName)) {
-      const currentViewState = this.options.appState.getCurrentViewState();
+    // TODO: seems better to check `currentViewState?.rel`, but
+    // redirect-idp form does not have `rel`
+    // device-apple-sso-extension form does have `rel` and still need redirect
+    const useFormRedirect = !currentViewState?.hasOwnProperty('method')
+      || currentViewState?.method?.toLowerCase() === 'get';
+    if (useFormRedirect) {
       Util.redirectWithFormGet(currentViewState.href);
       return;
     }
 
-    const modelJSON = model.toJSON();
-    const identifier = modelJSON.identifier;
-    if (identifier) {
-      // The callback function is passed two arguments:
-      // 1) username: The name entered by the user
-      // 2) operation: The type of operation the user is trying to perform:
-      //      - PRIMARY_AUTH
-      //      - FORGOT_PASSWORD
-      //      - UNLOCK_ACCOUNT
-      const operation = FORM_NAME_TO_OPERATION_MAP[formName];
-      modelJSON.identifier = this.settings.transformUsername(identifier, operation);
+    // Run hook: transform the user name (a.k.a identifier)
+    const modelJSON = this.transformIdentifier(formName, model);
+
+    // Error out when this is not a remediation form. Unexpected Exception.
+    const idx = this.options.appState.get('idx');
+    if (!this.options.appState.hasRemediationObject(formName)) {
+      this.options.settings.callGlobalError(`Cannot find http action for "${formName}".`);
```

Comment:
IIRC `callGlobalError` gets bubbled up to an unrecoverable error state to the end-user. Shouldn't this value (and the form error below) be localized?

Reply:
Good catch!!
Would the tool you or Nikhil developed be able to catch this issue?

Reply:
It won't - as our rules apply for only Courage templates and known components (`createCallout`, etc.).

Adding a lint rule for ensuring `callGlobalError` receives a `function` could help us resolve future instances of this! cc: @nikhilvenkatraman-okta

Reply:
Good callout. We can track localizing global errors separately https://oktainc.atlassian.net/browse/OKTA-383615